pkg/fake_pub_server/lib/fake_storage_server.dart

```diff
@@ -18,7 +20,12 @@ class FakeStorageServer {
     final server = await IOServer.bind('localhost', port);
     serveRequests(server.server, _handler);
     _logger.info('Storage server running on port $port');
-    await Future.delayed(Duration(days: 10000));
+    StreamSubscription sigsubs;
+    sigsubs = ProcessSignal.sighup.watch().listen((_) async {
+      await sigsubs.cancel();
+      await server.close();
+    });
+    await sigsubs.asFuture();
```

Comment:
Why not just:
```dart
await ProcessSignal.sighub.watch().first;
await server.close();
```

Also why not use `sigterm`? (I think it's the idiomatic one to use for terminating a process gracefully).